conan/tools/cmake/utils.py

```diff
@@ -22,9 +22,9 @@ def get_file_name(conanfile, forced_module_mode=None):
 def get_find_mode(conanfile):
     """
     :param conanfile: conanfile of the requirement
-    :return: "none" or "config" or "module" or "both" or None when not set
+    :return: "none" or "config" or "module" or "both" or "config" when not set
```

Comment:
This is in line with the documentation.

Note that if the packages explicitly use the python `None` value like this (instead of a string):
```self.cpp_info.set_property("cmake_find_mode", None)```

it would be mapped to `"config"` with this change - I wonder if we need to protect ourselves against that case, or if we should always assume that the property is always a string if defined.

Reply:
I don't think so. We try not to be overprotective, we end with a massive amount of checks that eventually make the things even worse 😓 So I wouldn't add a check here and maybe something to the documentation saying, please note that "none" has to be a string and not a `None` value that means "unset".

Reply:
Is "unset" a valid setting?  If so then I'd like to suggest to not allow "None" or any non-strings, and have "unset" an explicit setting that someone can see in the code and documentation.

If the parameter should be a string, then surely that should be a pre-assert?
You have to do manual checks when using a language that isn't statically typed ... that is the penalty of the "convenience" of a language like Python.

Especially if the consequence of someone typing None instead of "None" is several hours/days of debugging and struggling (surprise = bad), instead of a simple error message.